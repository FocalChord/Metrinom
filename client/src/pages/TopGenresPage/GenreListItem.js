import { Avatar, Box, Divider, ListItem, ListItemAvatar, ListItemText } from "@material-ui/core";
import CheckIcon from "@material-ui/icons/Check";
import React from "react";

const GenreListItem = ({ genre, selectGenre, checkIcon }) => {
    return (
        <Box>
            <ListItem button onClick={() => selectGenre(genre.rank)}>
                <ListItemAvatar>
                    <Avatar
                        style={{
                            borderRadius: 0,
                            width: 65,
                            height: 65,
                            color: "#1DB954",
                            backgroundColor: "rgba(0, 0, 0, 0)",
                        }}
                    >
                        {genre.rank + 1}
                    </Avatar>
                </ListItemAvatar>
                <ListItemText
                    style={{
                        paddingLeft: 20,
                    }}
                    primary={genre.nameUppercase}
                    secondary={
                        <React.Fragment>
                            {genre.number} Artist{genre.number > 1 ? "s" : ""}
                        </React.Fragment>
                    }
                />
                {genre.selected && <CheckIcon className={checkIcon} />}
            </ListItem>
            <Divider variant="inset" />
        </Box>
    );
};

export default GenreListItem;
