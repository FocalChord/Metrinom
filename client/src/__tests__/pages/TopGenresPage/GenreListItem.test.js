import React from "react";
import renderer from "react-test-renderer";
import "regenerator-runtime/runtime.js";
import GenreListItem from "../../../pages/TopGenresPage/GenreListItem";
import { MemoryRouter } from "react-router-dom";

test("<GenreListItem />", () => {
    const tree = renderer
        .create(
            <MemoryRouter>
                <GenreListItem genre={{ name: "hip hop" }} selectGenre={() => {}} checkIcon="check" />
            </MemoryRouter>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
