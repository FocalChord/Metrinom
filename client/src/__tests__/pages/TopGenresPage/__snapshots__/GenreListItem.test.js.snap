// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenreListItem /> 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <div
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemAvatar-root"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "borderRadius": 0,
            "color": "#1DB954",
            "height": 65,
            "width": 65,
          }
        }
      >
        NaN
      </div>
    </div>
    <div
      className="MuiListItemText-root"
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
         Artist
        
      </p>
    </div>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-inset"
  />
</div>
`;
